### DISCUSSION TITLE
For: prefer-rest-params, Possible addition under "When Not To Use It"

### DISCUSSION THREAD

**Discussion by clickwithclark** [eslint/eslint#15071] [Ideas]
I think it should not be used (or flagged if possible) when you are using default values in your parameter,
For example, the following would be flagged :
```js
  defaults.password = function (id, minLength = 6, maxLength = 15) {
    formInputs.push(id);
    initializeInput(id);
    setStrategy(id, defaultPasswordStrategy);
    formState.strategies[`${id}Args`] = [...arguments].slice(1);
    formState.strategies[id]();
  };
```
If I had to rewrite this to accommodate the rest parameter rule, I would have to fiddle with resulting indexes from the rest parameter defeating the purpose of having the default functionality. 

I welcome any guidance if I am seeing this incorrectly.

Cheers

**Comment by nzakas** [comment]
That’s a fair point. Feel free to create a PR to update the docs.
